### ISSUE TITLE
Modal aria-hidden elements should not be focussable Accessisbility Issue

### ISSUE THREAD

**Issue by taparikh** [OfficeDev/office-ui-fabric-react#9412]
## Codepen demonstrating the issue
 
https://codepen.io/taparikh/pen/WqbvBp

## Actual behavior:
When running the fast pass on a page with the modal popup opened, we hit the 'aria-elements should not be focusable' error.
Title: WCAG 4.1.2: ARIA hidden element must not contain focusable elements (div[data-is-visible="true"][aria-hidden="true"]:nth-child(1))
Tags: Accessibility, WCAG 4.1.2, aria-hidden-focus

Issue: ARIA hidden element must not contain focusable elements (aria-hidden-focus: https://dequeuniversity.com/rules/axe/3.2/aria-hidden-focus?application=msftAI)

Element path: div[data-is-visible="true"][aria-hidden="true"]:nth-child(1)

Snippet: <div tabindex="0" aria-hidden="true" data-is-visible="true" style="pointer-events: none; position: fixed;"></div>

How to fix:
Fix all of the following:
  Focusable content should have tabindex='-1' or be removed from the DOM

Despite applying the fix, we see the error repeatedly. When the modal is closed the error vanishes. 

Environment:
Chrome version 74.0.3729.169

This accessibility issue was found using Accessibility Insights for Web 2.4.1 (axe-core 3.2.2), a tool that helps find and fix accessibility issues. Get more information & download this tool at http://aka.ms/AccessibilityInsights.

## Expected behavior:
This error is invalid as we are not able to focus any elements outside the modal (which is expected). We are also not able to tab-out of the modal unless intentionally by pressing Enter on the 'X' modal button -  which is expected and the right behavior. The error seems misplaced.

## Documentation describing expected behavior
This error is invalid as we are not able to focus any elements outside the modal (which is expected). We are also not able to tab-out of the modal unless intentionally by pressing Enter on the 'X' modal button -  which is expected and the right behavior. The error seems misplaced.

![modal-popup](https://user-images.githubusercontent.com/51679982/59298565-a4f72780-8c3f-11e9-8c41-0778e10b23c3.jpg)

**Comment by natalieethell** [comment]
Hi @taparikh thanks for opening an issue! Unfortunately, your CodePen doesn't seem to be loading :( Is there any way you could check it out for us and provide us with one that loads/produces the error so we can help debug? Here's a modal example to get started: https://codepen.io/naethell/pen/VJYKGQ.

**Comment by taparikh** [comment]
Hi Natalie - it will be easier for me to demo it to you in our environment than create a codepen. This is because the modal is dependent on several dynamic parameters which cannot be replicated in a standalone codepen. 
Can I set up some time today for a quick call? @natalieethell

**Comment by natalieethell** [comment]
Re-opening the issue as @taparikh and I synced up offline. We're seeing a similar error from Accessibility Insights on the Fabric website coming from the flight tools element on the right side of the page.

![image](https://user-images.githubusercontent.com/14134000/60551253-900e3100-9cdf-11e9-80aa-906a3f1a6f71.png)

Notice in the screenshot below that I'm able to interact with the flight tools while the modal is open - I'm not sure if this is expected behavior, but it looks similar to what @taparikh is seeing.

![image](https://user-images.githubusercontent.com/14134000/60551360-fa26d600-9cdf-11e9-803a-d1c64b5f8bea.png)

It looks like the element with id "flightPicker" sits on the same level as the "main" element.

@jspurlin @dzearing do you have any ideas here as this relates to FocusTrapZone?

**Comment by taparikh** [comment]
@natalieethell any status update on this issue ?

**Comment by taparikh** [comment]
@Vitalius1 Hi! Just wanted to make sure that this issue is still being tracked and worked on! Let me know if you need more information.

**Comment by natalieethell** [comment]
@taparikh can you re-confirm what elements on your page are causing the error? On our website, the issue we're seeing is due to a "flightPicker" element that is controlled by developer.microsoft.com and sits on the same level as the "main" element in the DOM. I'm wondering if the placement of those elements causing the errors is part of the issue.

**Comment by taparikh** [comment]
Hi @natalieethell  Thanks for your response. 

Screenshot showing the position of the element using chrome browsers inspector tool. 
![image](https://user-images.githubusercontent.com/51679982/61335187-74c41b00-a7e1-11e9-9836-cebaa35da5a4.png)

Screenshot showing the accessibility error during FastPass of the Accessibility Insights tool.
![image](https://user-images.githubusercontent.com/51679982/61335058-f9fb0000-a7e0-11e9-9a61-f5fedf640c81.png)

Id be happy to hop on a call in case the above screenshots don't answer your question!

**Comment by natalieethell** [comment]
Hey @taparikh thanks for providing the screenshots! Could you provide what version of office-ui-fabric-react you're using?

**Comment by natalieethell** [comment]
After looking at this more, it looks like the problem is coming from FocusTrapZone's bumper element:

https://github.com/OfficeDev/office-ui-fabric-react/blob/f31e6a0fd26fdefc9d00df5a03f31153f1975620/packages/office-ui-fabric-react/src/components/FocusTrapZone/FocusTrapZone.tsx#L79-L86

https://github.com/OfficeDev/office-ui-fabric-react/blob/f31e6a0fd26fdefc9d00df5a03f31153f1975620/packages/office-ui-fabric-react/src/components/FocusTrapZone/FocusTrapZone.tsx#L98

It looks like the bumpers are tabbable when the focus trap zone is enabled (`disabled === false`), which is resulting in the tabbable element within the `aria-hidden` element. It looks like, however, you have an `aria-hidden` attribute on that element. I'm not seeing an `aria-hidden` attribute on our most recent version of FocusTrapZone, which is probably why I'm not seeing the error. Once we find out what version you're on, that should help us locate the root of the issue.

**Comment by taparikh** [comment]
@natalieethell The current version is office-ui-fabric-react "6.157.0"

**Comment by natalieethell** [comment]
Ahh okay, I would recommend upgrading. It looks like this Modal issue was resolved when removing aria-hidden from that element in FocusTrapZone here: https://github.com/OfficeDev/office-ui-fabric-react/pull/9019. Try upgrading to `office-ui-fabric-react@v6.180.0` or later and let me know if that solves your issue (it should)!

**Comment by taparikh** [comment]
Hi @natalieethell  Thanks for the fix ! It definitely passes the Fast Pass as you have described. Closing this issue with this comment !
